embc-app/Controllers/ReferralsController.cs

```diff
@@ -74,8 +76,13 @@ public async Task<IActionResult> Post(string registrationId, [FromBody] PostRequ
 
         [HttpPost("referralPdfs")]
         public async Task<IActionResult> GetReferralPdfs([FromBody] PrintReferrals printReferrals)
-        {
-            var content = $@"<!DOCTYPE html><html><body>This is a referral</body></html>";
+        {
+            var content = await referralsService.GetReferralHtmlPages(printReferrals);
+
+            if (string.IsNullOrWhiteSpace(content))
+            {
+                return NotFound(printReferrals.ReferralIds);
+            }
 
             return await pdfConverter.ConvertHtmlToPdfAsync(content);

```

Comment:
you ended up still having the logic in the contoller instead of pushing it down to the service, which if would be called 'ReferralPrintingService' would help you define the interface more SRP...

Reply:
Yes, I was planning on consolidating in all into the service, but this is still a WIP as I want to implement more improvements.  I just ran out of time, but just want to get this in DEV for Severin, so he can test his FE.